metabase/metabase
# Action Creator 8: Required Fields and Default Values

## Description

This allows form fields in action forms to be set as required. If they are not required, this allows a user to set a default value for that field.

## Demo

https://www.loom.com/share/ed58b57f729c48a69f03ee74c2326d87


## Testing Steps

- create a new action with at least 2 different form fields
- fields should be required by default, but mark one field as non-required and provide a default value for it.
- add that action to an app page (add a button, add a click action to your newly-created button and make it link to that particular action)
- exit edit mode and click your button to get an action form
- see that the save button is disabled until all required fields have values
- leave your non-required field blank and see that it saves your default value
- fill in your non-required field and see that it sets the provided (rather than the default) value
